client/src/utils.tsx

```diff
@@ -88,6 +92,23 @@ export default {
     }
   },
 
+  handleDialogEmailValidation: function(value, shouldValidate) {
+    if (!shouldValidate) {
+      return { isValid: true, message: null }
+    }
+    try {
+      const isValid = this.validateEmail(
+        value,
+        activeDomainNames,
+        wildcardActiveDomains
+      )
+      const message = isValid ? null : this.emailErrorMessage(activeDomainNames)
+      return { isValid, message }
+    } catch (e) {
+      return { isValid: false, message: <div>{e.message}</div> }
+    }
+  },
```

Comment:
This isn't very DRY, as this function is almost identical to `handleEmailValidation`. Suggest to introduce a common function:
```javascript
  _handleEmailValidation: function(
    value,
    validDomains,
    validWildcardDomains,
    shouldValidate
  ) {
    if (!shouldValidate) {
      return { isValid: true, message: null }
    }
    try {
      const isValid = this.validateEmail(
        value,
        validDomains,
        validWildcardDomains
      )
      const message = isValid ? null : this.emailErrorMessage(validDomains)
      return { isValid, message }
    } catch (e) {
      return { isValid: false, message: <div>{e.message}</div> }
    }
  },
```
and then replace the two existing functions with:
```javascript
  handleEmailValidation: function(value, shouldValidate) {
    return this._handleEmailValidation(
      value,
      domainNames,
      wildcardDomains,
      shouldValidate
    )
  },

  handleDialogEmailValidation: function(value, shouldValidate) {
    return this._handleEmailValidation(
      value,
      activeDomainNames,
      wildcardActiveDomains,
      shouldValidate
    )
  },
```